src/main/java/com/ibm/cloud/sdk/core/security/CloudPakForDataServiceAuthenticator.java

```diff
@@ -0,0 +1,399 @@
+/**
+ * (C) Copyright IBM Corp. 2019, 2021.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.ibm.cloud.sdk.core.security;
+
+import java.net.Proxy;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.ibm.cloud.sdk.core.http.RequestBuilder;
+
+/**
+ * This class provides an Authenticator implementation for the "CloudPakForData" environment.
+ * This authenticator will use the configured url and other properties to automatically fetch
+ * an access token for service to service authentication from the CloudPakForData service token service.
+ * When the access token expires, a new access token will be fetched.
+ *
+ * This authenticator uses the "GET /zen-data/internal/v1/service_token"
```

Comment:
```suggestion
 * This authenticator uses the "GET /v1/service_token"
```
Let's reference just the operation path here, since the context root should be configured as part of the "url" property.